### PR TITLE
Add Python 3 support

### PR DESCRIPTION
**Pull request by seanh**
Based on https://github.com/seanh/snapshotter/pull/24

### CODE REVIEW

File: snapshotter/snapshotter.py [hunk 1]
```diff
@@ -13,6 +17,21 @@
 import logging
 
 
+from snapshotter import PY2, PY3
+
+
+try:
```
Comment (funkyfuture) [inline snapshotter/snapshotter.py]: i'd put this into `__init__.py` as well.

File: snapshotter/snapshotter.py [hunk 2]
```diff
@@ -13,6 +17,21 @@
...
+    STDOUT_ENCODING = sys.stdout.encoding or sys.getdefaultencoding()
+except AttributeError:
+    STDOUT_ENCODING = sys.getdefaultencoding()
+
+
+if PY2:
+    text = unicode
+else:
```
Comment (funkyfuture) [inline snapshotter/snapshotter.py]: could be more explicit with `elif PY3:`.

**Comment by funkyfuture** [comment]
beside my remarks on more or less cosmetical issues, it lgtm.

i'll start testing it.

**Comment by funkyfuture** [comment]
btw, i discovered this module that looks interesting for testing: https://pypi.python.org/pypi/pyfakefs/2.4 resp. a more recent fork: https://github.com/jmcgeheeiv/pyfakefs

**Comment by seanh** [comment]
Thanks for the suggestions, I've committed them on master

**Pull request merged**